deposit-transactions.md

```diff
@@ -0,0 +1,165 @@
+
+This transaction MUST have the following values:
+
+1. `from` is the Depositor Account `0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001`.
+1. `to` is `0x4200000000000000000000000000000000000014` (the address of the L1 attributes predeploy
+   contract).
+1. `value` is `0`
+1. `data` is an abi encoded call to the [L1 Attributes Predeploy] contract's `setL1BlockValues()`
+   function with correct values associated with the corresponding L1 block.
+
+## Special Accounts on L2
+
+The L1 Attributes Deposit Transaction involves two special purpose accounts:
+
+1. The L1 Attributes Depositor Account
+2. The L1 Attributes Predeploy
+
+### L1 Attributes Depositor Account
+
+[l1-attributes-depositor-account]: #l1-attributes-depositor-account
+
+The Depositor Account is an EOA with no known private key. It has the address
+`0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001`. Its value returned by the `CALLER` and `ORIGIN`
+opcodes during execution of the L1 Attributes Deposit Transaction.
+
+### L1 Attributes Predeploy
+
+[l1-attributes-predeploy]: #l1-attributes-predeploy
+
+A predeployed contract on L2 at address `0x4200000000000000000000000000000000000014`, which holds
+certain block variables from the corresponding L1 block in storage, so that they may be accessed
+during the execution of the subsequent deposited transactions.
+
+The contract implements an authorization scheme, such that it only accepts state-changing calls from
+the [Depositor Account].
+
+The contract has the following solidity interface, and can be interacted with according to the
+[contract ABI specification][ABI].
+
+[ABI]: https://docs.soliditylang.org/en/v0.8.10/abi-spec.html
+
+```solidity
+interface L1BlockValues {
+
+  function setL1BlockValues(
+    uint256 number,
+    uint256 timestamp,
+    uint256 baseFee,
+    bytes32 hash
+  ) external;
+
+  function l1Number() view;
+  function l1Timestamp() view;
+  function l1BaseFee() view;
+  function l1Hash() view;
+}
+```
+
+## Deposited Transactions
+
+[deposited-transactions]: #deposited-transactions
+
+"Deposited Transactions" are derived from logs emitted by the [Deposit Feed
+contract][deposit-feed-contract] on L1.
+
+### Deposit Feed Contract
+
+[deposit-feed-contract]: #deposit-feed-contract
+
+The Deposit Feed contract is deployed to L1. Deposited Transactions are derived from the values in
+the `TransactionDeposited` event(s) emitted by the Deposit Feed contract.
+
+The Deposit Feed handles two special cases:
+
+1. A `to` value of `0`, which results in a contract creation on L2.
+2. A call from a contract account, in which case the `from` value is transformed to its L2 alias.
+   This prevents attacks in which a contract on L1 has the same address as a contract on L2 but
+   doesn't have the same code. We can safely ignore this for EOAs because they're guaranteed to have
+   the same "code" (i.e. no code at all). This also makes it possible for users to interact with
+   contracts on L2 even when the Sequencer is down.
```

Comment:
To clarify: we don't need to know about L2 state during block derivation to determine if the address is a contract or not, since it can be checked using the L1 during deposit-time instead?

Reply:
I think this works just based on L1 state, but I'm going to give this thought and respond properly.

Reply:
The assumption is that if an address is a contract on L1 it won't be a user on L2, and vice-versa (impossible in general to find collision etc). The only clash possible is L1 and L2 contracts with same address because of `CREATE2` with same constructor bytecode deploying two different contract bytecode. So yeah, we don't need L2 state.

Reply:
Explain what the L2 alias is here. Perhaps mention L2 contracts can nevertheless check for specific L1 and L2 contract callers.